### PR TITLE
Restore `markdownlint` tests

### LINKED ISSUE #12539: Restore markdownlint tests
`markdownlint` tests were removed in #10163 due to a security issue in a dependancy ([CVE-2019-10746](https://snyk.io/vuln/SNYK-JS-MIXINDEEP-450212)).

According to [snyk](https://snyk.io/test/npm/markdownlint/), `markdownlint` has no currently known vulnerabilities, so we should restore these tests.

### PR DESCRIPTION
**Pull request by xtqqczze**
# PR Summary

Fix #12539

## PR Context

`markdownlint` tests were removed in #10163 due to a security issue whoch has since been fixed in a newer version of `markdownlint`

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [x] None
    - **OR**
    - [ ] [Experimental feature(s) needed](https://github.com/MicrosoftDocs/PowerShell-Docs/blob/staging/reference/6/Microsoft.PowerShell.Core/About/about_Experimental_Features.md)
        - [ ] Experimental feature name(s): 
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [x] N/A or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
- **Tooling**
    - [x] I have considered the user experience from a tooling perspective and don't believe tooling will be impacted.
    - **OR**
    - [ ] I have considered the user experience from a tooling perspective and enumerated concerns in the summary. This may include:
        - Impact on [PowerShell Editor Services](https://github.com/PowerShell/PowerShellEditorServices) which is used in the [PowerShell extension](https://github.com/PowerShell/vscode-powershell) for VSCode (which runs in a different PS Host).
        - Impact on Completions (both in the console and in editors) - one of PowerShell's most powerful features.
        - Impact on [PSScriptAnalyzer](https://github.com/PowerShell/PSScriptAnalyzer) (which provides linting & formatting in the editor extensions).
        - Impact on [EditorSyntax](https://github.com/PowerShell/EditorSyntax) (which provides syntax highlighting with in VSCode, GitHub, and many other editors).

### CODE REVIEW

**Comment by xtqqczze** [comment]
Codacy markdown issues seem to contradict markdownlint issues.

**Review by iSazonov — DISMISSED** [review]

File: test/common/markdown/gulpfile.js [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+        }
+
+        var rootJsonFile = rootpath + "/.markdownlint.json";
+        var fs = require("fs");
+        fs.appendFileSync("markdownissues.txt", "--EMPTY--\r\n");
+        return gulp.src(paths, { "read": false })
+            .pipe(through2.obj(function obj(file, enc, next) {
+                markdownlint({
+                        "files": [file.path],
+                        "config": require(rootJsonFile)
```
Comment (xtqqczze) [inline test/common/markdown/gulpfile.js:42]: > Found non-literal argument in require (security/detect-non-literal-require)

**Comment by xtqqczze** [comment]
```
┌───────────────┬──────────────────────────────────────────────────────────────┐
│ low           │ Prototype Pollution                                          │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Package       │ yargs-parser                                                 │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Patched in    │ >=13.1.2 <14.0.0 || >=15.0.1 <16.0.0 || >=18.1.2             │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Dependency of │ c8                                                           │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Path          │ c8 > yargs-parser                                            │
├───────────────┼──────────────────────────────────────────────────────────────┤
└───────────────┴──────────────────────────────────────────────────────────────┘
┌───────────────┬──────────────────────────────────────────────────────────────┐
│ low           │ Prototype Pollution                                          │
├───────────────┼──────────────────────────────────────────────────────────────┤
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Patched in    │ >=13.1.2 <14.0.0 || >=15.0.1 <16.0.0 || >=18.1.2             │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Dependency of │ cpy-cli                                                      │
│ Path          │ cpy-cli > meow > yargs-parser                                │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ More info     │ https://www.npmjs.com/advisories/1500                        │
└───────────────┴──────────────────────────────────────────────────────────────┘
```

**Comment by xtqqczze** [comment]
```
┌───────────────┬──────────────────────────────────────────────────────────────┐
│ low           │ Prototype Pollution                                          │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Package       │ yargs-parser                                                 │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Patched in    │ >=13.1.2 <14.0.0 || >=15.0.1 <16.0.0 || >=18.1.2             │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Path          │ gulp > gulp-cli > yargs > yargs-parser                       │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ More info     │ https://www.npmjs.com/advisories/1500                        │
└───────────────┴──────────────────────────────────────────────────────────────┘
```

File: tools/install-powershell.sh-README.md [hunk 2]
```diff
@@ -66,29 +66,3 @@ bash <(wget -O - https://raw.githubusercontent.com/PowerShell/PowerShell/master/
 ### Installation To do list
 
 * Detect and wait when package manager is busy/locked? - at least Ubuntu (CentOS does this internally)
-
-# install-powershell.ps1
```
Comment (TravisEz13) [inline tools/install-powershell.sh-README.md:70]: renaming the file make break links in the repo.  they may only be found when we merge to master.

**Comment by TravisEz13** [comment]
There mere presence of a high scored vuln on the machine will trigger the same issue that cause this to be removed.

**Comment by TravisEz13** [comment]
I made a couple of edits to your branch.  Please pull your branch if you need to edit something.

**Comment by TravisEz13** [comment]
That component is not currently generating an alert.

**Comment by xtqqczze** [comment]
The maintainer of `gulp` says the vulnerable dependency will not be updated until the project  releases a new major version, which does not appear to be imminient.

See: https://github.com/yargs/yargs-parser/issues/264#issuecomment-622955036

**Comment by xtqqczze** [comment]
@TravisEz13 Perhaps we could do without `gulp` anyway, and use  [`npm-run-script`](https://docs.npmjs.com/cli/run-script) or equivalent?

File: tools/install-powershell.sh-README.md (hunk 2 above)
Comment (xtqqczze) [reply tools/install-powershell.sh-README.md:70]: Full text search for `install-powershell-readme.md` finds no matches in the repo,

**Comment by TravisEz13** [comment]
@xtqqczze  I'm not tied to gulp.  As long as we still get test results.  But if it reduces dependencies, I'm all for it.

**Comment by TravisEz13** [comment]
Tell me if you want me to merge this as is and then update, or you want to change it first.

**Comment by xtqqczze** [comment]
@TravisEz13 Merge this for now please.

File: test/common/markdown/markdown.tests.ps1 [hunk 3]
```diff
@@ -0,0 +1,98 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
```
Comment (TravisEz13) [inline test/common/markdown/markdown.tests.ps1:1]:
```suggestion
# Copyright (c) Microsoft Corporation.
```

File: test/common/markdown/gulpfile.js [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+                        }
+                        next(err, file);
+                    });
+            }))
+            .pipe(concat("markdownissues.txt", { newLine: "\r\n" }))
+            .pipe(gulp.dest("."));
+    });
+}
+
+runTest();
```
Comment (TravisEz13) [inline test/common/markdown/gulpfile.js:57]:
```suggestion
runTest();

```

**Comment by xtqqczze** [comment]
`PowerShell-CI-static-analysis` failure is unrelated, issue opened: #12775

**Comment by xtqqczze** [comment]
force pushed to restart CI

**Comment by TravisEz13** [comment]
Please resolve conflicts

**Comment by xtqqczze** [comment]
rebased to resolve conflicts

**Comment by xtqqczze** [comment]
@TravisEz13 I'm not my change in 4180235 to use `Write-Host` is the best approach, but I think we should log `$markdownErrors` somehow...

**Comment by xtqqczze** [comment]
rebased to fix additionally markdownlint errors

File: test/common/markdown/gulpfile.js [hunk 5]
```diff
@@ -0,0 +1,57 @@
+function runTest() {
```
Comment (TravisEz13) [inline test/common/markdown/gulpfile.js:1]: This should have a copyright

**Review by TravisEz13 — CHANGES_REQUESTED** [review]
One one comment to address

File: test/common/markdown/gulpfile.js (hunk 5 above)
Comment (TravisEz13) [reply test/common/markdown/gulpfile.js:1]:
```suggestion
// Copyright (c) Microsoft Corporation.
// Licensed under the MIT License.

function runTest() {
```

**Review by TravisEz13 — APPROVED** [review]

**Comment by TravisEz13** [comment]
@PoshChan Please remind me in 1 hour

**Comment by PoshChan** [comment]
@TravisEz13, this is the reminder you requested 1 hour ago

**Pull request merged**